apps/cnspec/cmd/vuln.go

```diff
@@ -59,15 +59,14 @@ var vulnCmdRun = func(cmd *cobra.Command, runtime *providers.Runtime, cliRes *pl
 		log.Fatal().Msg(unauthedErrorMsg)
 	}
 
-	res, err := runtime.Provider.Instance.Plugin.Connect(&plugin.ConnectReq{
+	err = runtime.Connect(&plugin.ConnectReq{
 		Features: conf.Features,
 		Asset:    cliRes.Asset,
 		Upstream: runtime.UpstreamConfig,
-	}, nil)
+	})
 	if err != nil {
 		log.Fatal().Err(err).Msg("could not load asset information")
```

Comment:
can we remove the fatal here and follow the same pattern as here https://github.com/mondoohq/cnspec/pull/900. We want to exit properly instead of panicking

Reply:
Not that easy. I gave it a try. The current structs do not support it. We would need to change cnquery, and than preferably also all the other commands associated with the struct: `shell`, `run`, and `scan`.

That would blow up this PR and log.Fatal already exits with `os.Exit(1)` and does not panic.

Reply:
alright, let's leave that out. Need to fix the tests though

Reply:
Fixed